python/nx-cugraph/nx_cugraph/algorithms/core.py

```diff
@@ -0,0 +1,91 @@
+# Copyright (c) 2023, NVIDIA CORPORATION.
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import cupy as cp
+import networkx as nx
+import numpy as np
+import pylibcugraph as plc
+
+import nx_cugraph as nxcg
+from nx_cugraph.utils import networkx_algorithm, not_implemented_for
+
+__all__ = ["k_truss"]
+
+
+@not_implemented_for("directed")
+@not_implemented_for("multigraph")
+@networkx_algorithm
+def k_truss(G, k):
+    if is_nx := isinstance(G, nx.Graph):
+        G = nxcg.from_networkx(G, preserve_all_attrs=True)
+    if nxcg.number_of_selfloops(G) > 0:
+        raise nx.NetworkXError(
+            "Input graph has self loops which is not permitted; "
+            "Consider using G.remove_edges_from(nx.selfloop_edges(G))."
+        )
+    # TODO: create renumbering helper function(s)
+    if k < 3:
+        # Drop nodes with zero degree
+        degrees = G._degrees_array()
+        # Renumber step 0: node indices
+        node_indices = degrees.nonzero()[0]
+        if degrees.size == node_indices.size:
+            # No change
+            return G if is_nx else G.copy()
+        src_indices = G.src_indices
+        dst_indices = G.dst_indices
+        # Renumber step 1: edge values (no changes needed)
+        edge_values = {key: val.copy() for key, val in G.edge_values.items()}
+        edge_masks = {key: val.copy() for key, val in G.edge_masks.items()}
+    else:
+        # int dtype for edge_indices would be preferred
+        edge_indices = cp.arange(G.src_indices.size, dtype=np.float64)
```

Comment:
Does PLC only handle edge weights with float dtypes?

Reply:
Yes. Only float32 and float64. This is handled more generally in #3954:
https://github.com/rapidsai/cugraph/pull/3954/files#diff-b87ebafc61a8ff368beff74819475091903f5c7590ae4a7d56363fb5ed40c1a9